### ISSUE TITLE
"timed out waiting for helper registration" while downloading 12.6.3 Monterey installer

### ISSUE THREAD

**Issue by UNIVAC-Colonel-Panic** [munki/macadmin-scripts#119]
I'm running macOS 11.7.4. 
~~~
(snip)
Downloading http://swcdn.apple.com/content/downloads/17/14/032-33828-A_KX25OQBX8S/lz9yip4rmenyju8nkv4wpaqtdmvzfogkcb/InstallInfo.plist...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0   188    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Making empty sparseimage...
installer: Package name is macOS Monterey
installer: Installing at base path /private/tmp/dmg.L0xS86
installer: The install was successful.
*********************************************************
*** Working around a very dumb Apple bug in a package ***
*** postinstall script that fails to correctly target ***
*** the Install macOS.app when installed to a volume  ***
*** other than the current boot volume.               ***
***       Please file feedback with Apple!            ***
*********************************************************
Product downloaded and installed to /Volumes/apfs-100gb-dynamic/Install_macOS_12.6.3-21G419.sparseimage
Making read-only compressed disk image containing Install macOS Monterey.app...
........................................................................................................2023-03-25 15:41:51.277 hdiutil[8394:330543] -[DIHelperProxy(Thread) waitForHelperDone] timed out waiting for helper registration
.......................
created: /Volumes/apfs-100gb-dynamic/Install_macOS_12.6.3-21G419.dmg
Disk image created at: ./Install_macOS_12.6.3-21G419.dmg
~~~

There is plenty of disk space available on root i.e. the startup disk (41gb) and on /Volumes/apfs-100gb-dynamic: (45gb).

**Comment by gregneagle** [comment]
No other reports of this. I've created Big Sur, Monterey, and Ventura installer dmgs several times since this was posted. Is this reproducible?

**Comment by UNIVAC-Colonel-Panic** [comment]
thanks for the response. it had been reproducible several times when i initially reported it. i will try again and report back in a day or two.

**Comment by gregneagle** [comment]
I wonder if this is a bad interaction with some locally-installed security software -- something that is interfering with the mounting or creation of disk images.

**Comment by UNIVAC-Colonel-Panic** [comment]
FYI, a couple of weeks ago my Mac upgraded from macOS 11.7.4 to 11.7.8.

The download for the 12.6.3 Monterey installer is no longer an option, however, I have finished downloading 12.6.4 to 12.6.7. The hdiutil error: 

`[DIHelperProxy(Thread) waitForHelperDone] timed out waiting for helper registration` 

happened for every one, so yeah it is consistently reproducible. It happens around 30 minutes into the whole process.

I carefully monitored disk space available for each version. The startup disk (root file system) stayed at about 50g free. I downloaded these installers to a separate filesystem and never let the space available get below about 32g. I didn't realize the 12g InstallAssistant.pkg stays around after the python script finishes. Here's the one for 12.6.7 that I found in the directory **content/downloads/18/30/032-97690-A_RNQLC09SLR/rd7o4w50xrst3ofw3bhlbkjypi24eldqop**:

`-rw-r--r--  1 root  admin  12406425078 Jul 14 18:01 InstallAssistant.pkg`

When disk space available is too low, I have seen this be the cause for a different hdiutil error:

`Command '[u'/usr/bin/hdiutil', u'create', u'-fs', u'HFS+', u'-srcfolder', u'/private/tmp/dmg.lIqgj4/Applications/Install macOS Monterey.app', u'./Install_macOS_12.6.4-21G526.dmg']' returned non-zero exit status 1`

There is no locally-installed security software that I'm aware of but I can see why that could be a factor.

So bottom line, the "timed out" error is consistently reproducible and doesn't appear to be related to disk space available at all.

**Comment by gregneagle** [comment]
I have yet to see this issue ever, and in the last week I built installer dmgs for the latest Big Sur, Monterey, and Ventura versions as well as Beta 4 of Sonoma. The machine I'm using is an iMac Pro running Monterey. Maybe this is a Ventura issue I have not yet seen?

**Comment by gregneagle** [comment]
Also interesting that the original r/w non-compressed disk image succeeds:
`created: /Users/<me>/macadmin-scripts/Install_macOS_13.5-22G74.dmg`
But the conversion to read-only compressed disk image is what failed (at least for @volsk)

**Comment by gregneagle** [comment]
Still working as expected here:
```
Making empty sparseimage...
installer: Package name is macOS Sonoma beta
installer: Installing at base path /private/tmp/dmg.N67e0Q
installer: The install was successful.
*********************************************************
*** Working around a very dumb Apple bug in a package ***
*** postinstall script that fails to correctly target ***
*** the Install macOS.app when installed to a volume  ***
*** other than the current boot volume.               ***
***       Please file feedback with Apple!            ***
*********************************************************
Product downloaded and installed to /Volumes/Catalina/Users/Shared/macadmin-scripts/Install_macOS_14.0-23A5328b.sparseimage
Making read-only compressed disk image containing Install macOS Sonoma beta.app...
...............................................................................................................................
created: /Volumes/Catalina/Users/Shared/macadmin-scripts/Install_macOS_14.0-23A5328b.dmg
Disk image created at: ./Install_macOS_14.0-23A5328b.dmg
```

**Comment by Excremation** [comment]
I'm also experiencing this creating the dmg for Sonoma 14.4.1.

![image](https://github.com/munki/macadmin-scripts/assets/136895206/5719e421-cbbb-4116-93bd-464f7ff11a0c)

**Comment by gregneagle** [comment]
Again, I don't think there's anything reproducible/actionable here.

**Comment by joelcarlton** [comment]
I had this issue too
